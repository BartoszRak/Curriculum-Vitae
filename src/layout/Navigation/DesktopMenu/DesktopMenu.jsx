import React from 'react'
import PropTypes from 'prop-types'
import { withStyles, Button } from '@material-ui/core'

import { useScrollTo } from '~hooks'

import style from './DesktopMenu.style'

export function DesktopMenu({ classes, routes }) {
  const { scrollTo } = useScrollTo()
  return (
    <React.Fragment>
      {routes.map(route => (
        <Button
          classes={{
            label: classes.buttonLabel,
            root: classes.button,
          }}
          color="secondary"
          key={route.name}
          onClick={() => scrollTo(route.elementId)}
        >
          {route.name}
        </Button>
      ))}
    </React.Fragment>
  )
}

DesktopMenu.propTypes = {
  classes: PropTypes.object,
  routes: PropTypes.arrayOf(PropTypes.shape({
    elementId: PropTypes.string,
    name: PropTypes.string,
  })).isRequired,
}

export default withStyles(style)(DesktopMenu)
