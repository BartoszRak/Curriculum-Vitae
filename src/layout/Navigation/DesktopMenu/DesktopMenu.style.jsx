export default ({ palette, spacing }) => ({
  button: {
    '&:hover': {
      background: 'rgba(0, 0, 0, 0.05)',
    },
    borderRadius: 100,
    marginRight: spacing.unit * 4,
    padding: `${spacing.unit * 0.5}px ${spacing.unit * 2}px`,
  },
  buttonLabel: {
    color: palette.secondary.light,
    fontSize: '1rem',
    fontWeight: 'normal',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.8)',
  },
})
