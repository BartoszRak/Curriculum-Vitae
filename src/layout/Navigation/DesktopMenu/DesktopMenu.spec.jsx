import React from 'react'
import { shallow } from 'enzyme'

import { DesktopMenu } from './DesktopMenu'

const initComponent = overrides => {
  const mockProps = {
    classes: {},
    routes: [
      {
        elementId: 'testElementId1',
        name: 'testElementName1',
      },
      {
        elementId: 'testElementId2',
        name: 'testElementName2',
      },
    ],
  }
  const mockMethods = {}
  const wrapper = shallow(
    <DesktopMenu {...mockProps} {...mockMethods} {...overrides} />
  )
  return { mockProps, wrapper }
}

describe('global: DesktopMenu', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
