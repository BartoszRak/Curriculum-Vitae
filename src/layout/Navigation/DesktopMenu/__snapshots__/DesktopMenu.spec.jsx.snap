// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: DesktopMenu should render as expected 1`] = `
<Fragment>
  <ButtonMock
    classes={
      Object {
        "label": undefined,
        "root": undefined,
      }
    }
    color="secondary"
    key="testElementName1"
    onClick={[Function]}
  >
    testElementName1
  </ButtonMock>
  <ButtonMock
    classes={
      Object {
        "label": undefined,
        "root": undefined,
      }
    }
    color="secondary"
    key="testElementName2"
    onClick={[Function]}
  >
    testElementName2
  </ButtonMock>
</Fragment>
`;
